### PR TITLE
Use labels.FromString abstraction

### PR DESCRIPTION
**Pull request by bboreham**
#### What this PR does

Use `labels.FromString` abstraction instead of assuming `labels.Labels` is a slice that we can construct directly. This will allow flexibility in the implementation.

All but one of the changes are in tests; none are performance-sensitive.

Similar PR upstream: https://github.com/prometheus/prometheus/pull/11048

#### Checklist

- [x] Tests updated
- NA Documentation added
- NA `CHANGELOG.md` updated - not user-visible.

### CODE REVIEW

**Review by colega — APPROVED** [review]
LGTM, thanks. I wish we could use `faillint` to lint the instantiation somehow, but apparently it's not possible.

**Comment by colega** [comment]
Maybe we could consider introducing at least this simple grep? It found some more usages:

```
$ grep -R 'labels.Labels{{' ./pkg
./pkg/distributor/distributor_test.go:			labels:   []labels.Labels{{{Name: labels.MetricName, Value: "testmetric"}, {Name: "foo", Value: "bar"}}},
./pkg/distributor/distributor_test.go:			expectedErr:        fmt.Sprintf("received an exemplar with no valid labels, timestamp: %d series: %+v labels: {}", now, labels.Labels{{Name: labels.MetricName, Value: "testmetric"}, {Name: "foo", Value: "bar"}}),
./pkg/storage/tsdb/upload_test.go:	}, 100, 0, 1000, labels.Labels{{Name: "ext1", Value: "val1"}}, 124, metadata.NoneFunc)
```

**Comment by pracucci** [comment]
> Maybe we could consider introducing at least this simple grep as part of `make lint`? It found some more usages:

I don't think it's necessary. The goal is to get https://github.com/prometheus/prometheus/pull/10887 merged. After it will get merged, any attempt to use `labels.Labels` as a slice will result in a compilation error. This PR is just a preliminary refactoring. If we'll use `labels.Labels` as a slice again in the near future, we'll just fix it once upgrading the upstream Prometheus after PR 10887 is merged.

**Review by pracucci — APPROVED** [review]
LGTM, thanks! (modulo a nit I'm going to auto-commit if you don't mind)

**Pull request merged**
